src/transformStateWithClones.js

```diff
@@ -7,7 +7,36 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const resultArray = [];
+
+  for (const action of actions) {
+    const { type } = action;
+    const currentState = { ...resultArray[resultArray.length - 1] || state };
```

Comment:
This line may cause performance issues when the length of 'resultArray' increases, as accessing the last element of an array with the 'length - 1' approach is expensive. Consider keeping a separate variable for the 'currentState' that gets updated with each action.